testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -381,55 +381,76 @@ public void testTrinoCreatesSparkDrops()
     @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS, ICEBERG_REST}, dataProvider = "storageFormats")
     public void testSparkReadsTrinoPartitionedTable(StorageFormat storageFormat)
     {
+        if (storageFormat == StorageFormat.ORC) {
+            throw new SkipException("Open Spark issue https://github.com/apache/iceberg/issues/3139 to read ORC table with nested partition column.");
+        }
         String baseTableName = toLowerCase("test_spark_reads_trino_partitioned_table_" + storageFormat);
         String trinoTableName = trinoTableName(baseTableName);
         String sparkTableName = sparkTableName(baseTableName);
         onTrino().executeQuery("DROP TABLE IF EXISTS " + trinoTableName);
 
-        onTrino().executeQuery(format("CREATE TABLE %s (_string VARCHAR, _varbinary VARBINARY, _bigint BIGINT) WITH (partitioning = ARRAY['_string', '_varbinary'], format = '%s')", trinoTableName, storageFormat));
-        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', X'0ff102f0feff', 1001), ('b', X'0ff102f0fefe', 1002), ('c', X'0ff102fdfeff', 1003)", trinoTableName));
+        onTrino().executeQuery(format("CREATE TABLE %s (_string VARCHAR, _varbinary VARBINARY, _bigint BIGINT, _struct ROW(_field BIGINT)) WITH (partitioning = ARRAY['_string', '_varbinary', '\"_struct._field\"'], format = '%s')", trinoTableName, storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', X'0ff102f0feff', 1001, ROW(1)), ('b', X'0ff102f0fefe', 1002, ROW(2)), ('c', X'0ff102fdfeff', 1003, ROW(3))", trinoTableName));
 
-        Row row1 = row("b", new byte[] {15, -15, 2, -16, -2, -2}, 1002);
-        String selectByString = "SELECT * FROM %s WHERE _string = 'b'";
+        Row row1 = row("b", new byte[]{15, -15, 2, -16, -2, -2}, 1002, 2);
+        String selectByString = "SELECT _string, _varbinary, _bigint, _struct._field FROM %s WHERE _string = 'b'";
         assertThat(onTrino().executeQuery(format(selectByString, trinoTableName)))
                 .containsOnly(row1);
+        // TODO
```

Comment:
TODO task. Parquet format returning null for partition nested field.

Reply:
If Spark Iceberg can't handle this kind of use-case please see whether there is already reported an issue to address this missing functionality.
We shouldn't introduce `TODO`s in the code relating to Spark limitations. Trino does not depend on Spark.

Simply add an assertion that the query is failing with an expected message.

Reply:
I could not find any existing issue for `PARQUET` issue. Where Saprk returns `null` for the partitioned nested field.